core/app/[locale]/(default)/(faceted)/category/[slug]/page.tsx

```diff
@@ -1,42 +1,63 @@
+import { getFormatter, getTranslations, setRequestLocale } from 'next-intl/server';
+import { createSearchParamsCache } from 'nuqs/server';
+import { cache } from 'react';
+
+import { Stream } from '@/vibes/soul/lib/streamable';
+import { Breadcrumb } from '@/vibes/soul/primitives/breadcrumbs';
+import { CursorPaginationInfo } from '@/vibes/soul/primitives/cursor-pagination';
+import { ListProduct } from '@/vibes/soul/primitives/products-list';
+import { ProductsListSection } from '@/vibes/soul/sections/products-list-section';
+import { getFilterParsers } from '@/vibes/soul/sections/products-list-section/filter-parsers';
+import { Filter } from '@/vibes/soul/sections/products-list-section/filters-panel';
+import { Option as SortOption } from '@/vibes/soul/sections/products-list-section/sorting';
+import { facetsTransformer } from '~/data-transformers/facets-transformer';
+import { pricesTransformer } from '~/data-transformers/prices-transformer';
 
-import { Breadcrumbs } from '~/components/breadcrumbs';
-import { ProductCard } from '~/components/product-card';
-import { Pagination } from '~/components/ui/pagination';
-
-import { FacetedSearch } from '../../_components/faceted-search';
-import { MobileSideNav } from '../../_components/mobile-side-nav';
-import { SortBy } from '../../_components/sort-by';
 import { fetchFacetedSearch } from '../../fetch-faceted-search';
 
 import { CategoryViewed } from './_components/category-viewed';
-import { EmptyState } from './_components/empty-state';
-import { SubCategories } from './_components/sub-categories';
 import { getCategoryPageData } from './page-data';
 
+type SearchParams = Record<string, string | string[] | undefined>;
+
 interface Props {
   params: Promise<{
     slug: string;
     locale: string;
   }>;
-  searchParams: Promise<Record<string, string | string[] | undefined>>;
+  searchParams: Promise<SearchParams>;
 }
 
-export async function generateMetadata({ params }: Props): Promise<Metadata> {
-  const { slug } = await params;
-  const categoryId = Number(slug);
-
-  const data = await getCategoryPageData({
-    categoryId,
+const createFiltersSearchParamCache = cache(async (categoryId: number) => {
+  const search = await fetchFacetedSearch({ category: categoryId });
+  const facets = search.facets.items.filter((facet) => facet.__typename !== 'CategorySearchFilter');
+  const transformedFacets = await facetsTransformer({
+    refinedFacets: facets,
+    allFacets: facets,
+    searchParams: {},
   });
+  const filters = transformedFacets.filter((facet) => facet != null);
+
+  return createSearchParamsCache(getFilterParsers(filters));
+});
+
+async function getCategory(categoryId: number) {
+  const data = await getCategoryPageData({ categoryId });
 
   const category = data.category;
 
-  if (!category) {
-    return {};
-  }
+  if (category == null) notFound();
+
+  return category;
+}
+
+export async function generateMetadata({ params }: Props): Promise<Metadata> {
```

Comment:
🍹 Can we move this function closer to Page component?

Reply:
Done!